src/services/auth.hooks.ts

```diff
@@ -29,11 +30,17 @@ export const useProvideAuth = () => {
     setUser(undefined);
   };
 
+  const updateLoadingState = () => {
+    setIsLoading(serverSignin.isLoading);
+    return isLoading;
+  };
+
   return {
     user,
     devSignin,
     signin,
-    signout
+    signout,
+    updateLoadingState
```

Comment:
Why expose an update function instead of directly exposing the `isLoading` field?